### ISSUE TITLE
pm2 5.0.0 Cannot find module 'pm2-sysmonit'

### ISSUE THREAD

**Issue by xqin** [Unitech/pm2#5091]
## What's going wrong?

```
internal/modules/cjs/loader.js:796
    throw err;
    ^

Error: Cannot find module 'pm2-sysmonit'
Require stack:
- /home/jenkins/a/node_modules/pm2/lib/API/Extra.js
- /home/jenkins/a/node_modules/pm2/lib/API.js
- /home/jenkins/a/node_modules/pm2/lib/binaries/CLI.js
- /home/jenkins/a/node_modules/pm2/bin/pm2
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:793:17)
    at Function.resolve (internal/modules/cjs/helpers.js:80:19)
    at API.CLI.launchSysMonitoring (/home/jenkins/a/node_modules/pm2/lib/API/Extra.js:38:41)
    at /home/jenkins/a/node_modules/pm2/lib/API.js:184:12
    at /home/jenkins/a/node_modules/pm2/lib/Client.js:114:16
    at Timeout._onTimeout (/home/jenkins/a/node_modules/pm2/lib/Client.js:372:9)
    at listOnTimeout (internal/timers.js:531:17)
    at processTimers (internal/timers.js:475:7) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/jenkins/a/node_modules/pm2/lib/API/Extra.js',
    '/home/jenkins/a/node_modules/pm2/lib/API.js',
    '/home/jenkins/a/node_modules/pm2/lib/binaries/CLI.js',
    '/home/jenkins/a/node_modules/pm2/bin/pm2'
  ]
}
```

As an optional module, when it is not installed, the code should not report an error and exit.

In the previous version `4.5.6`, the same operation, pm2 can run normally, and there is no error.

## How could we reproduce this issue?

```
npm install --no-optional pm2

npx pm2 list
```

## Supporting information

```
--- PM2 report ----------------------------------------------------------------
Date                 : Tue Jun 08 2021 15:41:42 GMT+0800 (China Standard Time)
===============================================================================
--- Daemon --------------------------------------------------------------------
pm2d version         : 5.0.0
node version         : 12.14.1
node path            : /usr/local/bin/npx
argv                 : /usr/local/bin/node,/home/jenkins/a/node_modules/pm2/lib/Daemon.js
argv0                : node
user                 : undefined
uid                  : 1001
gid                  : 1001
uptime               : 1min
===============================================================================
--- CLI -----------------------------------------------------------------------
local pm2            : 5.0.0
node version         : 12.14.1
node path            : /usr/local/bin/npx
argv                 : /usr/local/bin/node,/home/jenkins/a/node_modules/.bin/pm2,report
argv0                : node
user                 : undefined
uid                  : 1001
gid                  : 1001
===============================================================================
--- System info ---------------------------------------------------------------
arch                 : x64
platform             : linux
type                 : Linux
cpus                 : Intel Core Processor (Haswell)
cpus nb              : 8
freemem              : 337178624
totalmem             : 25281744896
home                 : /home/jenkins
===============================================================================
--- PM2 list ------------------------------------------------------------------
┌─────┬───────────┬─────────────┬─────────┬─────────┬──────────┬────────┬──────┬───────────┬──────────┬──────────┬──────────┬──────────┐
│ id  │ name      │ namespace   │ version │ mode    │ pid      │ uptime │ ↺    │ status    │ cpu      │ mem      │ user     │ watching │
└─────┴───────────┴─────────────┴─────────┴─────────┴──────────┴────────┴──────┴───────────┴──────────┴──────────┴──────────┴──────────┘
===============================================================================
--- Daemon logs ---------------------------------------------------------------
/home/jenkins/.pm2/pm2.log last 20 lines:
PM2        | 2021-06-08T15:39:51: PM2 log: ===============================================================================
PM2        | 2021-06-08T15:39:51: PM2 log: --- New PM2 Daemon started ----------------------------------------------------
PM2        | 2021-06-08T15:39:51: PM2 log: Time                 : Tue Jun 08 2021 15:39:51 GMT+0800 (China Standard Time)
PM2        | 2021-06-08T15:39:51: PM2 log: PM2 version          : 5.0.0
PM2        | 2021-06-08T15:39:51: PM2 log: Node.js version      : 12.14.1
PM2        | 2021-06-08T15:39:51: PM2 log: Current arch         : x64
PM2        | 2021-06-08T15:39:51: PM2 log: PM2 home             : /home/jenkins/.pm2
PM2        | 2021-06-08T15:39:51: PM2 log: PM2 PID file         : /home/jenkins/.pm2/pm2.pid
PM2        | 2021-06-08T15:39:51: PM2 log: RPC socket file      : /home/jenkins/.pm2/rpc.sock
PM2        | 2021-06-08T15:39:51: PM2 log: BUS socket file      : /home/jenkins/.pm2/pub.sock
PM2        | 2021-06-08T15:39:51: PM2 log: Application log path : /home/jenkins/.pm2/logs
PM2        | 2021-06-08T15:39:51: PM2 log: Worker Interval      : 30000
PM2        | 2021-06-08T15:39:51: PM2 log: Process dump file    : /home/jenkins/.pm2/dump.pm2
PM2        | 2021-06-08T15:39:51: PM2 log: Concurrent actions   : 2
PM2        | 2021-06-08T15:39:51: PM2 log: SIGTERM timeout      : 1600
PM2        | 2021-06-08T15:39:51: PM2 log: ===============================================================================

```

**Comment by xqin** [comment]
bug from:   https://github.com/Unitech/pm2/commit/1def3d8fb96823ab157b62d3b677f528a2223f48

**Comment by Unitech** [comment]
Working on it
Could you for now install pm2 without the flag —no-optional

**Comment by xqin** [comment]
Currently, I have locked version with 4.5.6

**Comment by Unitech** [comment]
Fixed via pm2@5.0.1
Feel free to reopen an issue if you see anything

**Comment by xqin** [comment]
According to the analysis of the place where launchSysMonitoring is called in the code, it seems that they don't care about callback, so why not remove the logic of callback.


![image](https://user-images.githubusercontent.com/1265888/121157403-1d6b4000-c87c-11eb-918b-2e6ac883f347.png)


and   recommended to adjust the order of the codes as below.

![image](https://user-images.githubusercontent.com/1265888/121157613-4d1a4800-c87c-11eb-8fdb-318f7376a46d.png)
